lib/src/checkBox/checkBox.component.ts

```diff
@@ -0,0 +1,78 @@
+import { Component, Input, forwardRef, ViewEncapsulation, OnInit, TemplateRef, Output, EventEmitter, Provider } from "@angular/core";
+import { NG_VALUE_ACCESSOR, ControlValueAccessor } from "@angular/forms";
+import { randomId } from "@sebgroup/frontend-tools/dist/randomId";
+
+export const CUSTOM_CHECKBOX_CONTROL_VALUE_ACCESSOR: Provider = {
+    provide: NG_VALUE_ACCESSOR,
+    useExisting: forwardRef(() => CheckBoxComponent),
+    multi: true,
+};
+
+@Component({
+    selector: "sebng-checkbox",
+    templateUrl: "./checkBox.component.html",
+    styleUrls: ["./checkBox.component.scss"],
+    providers: [CUSTOM_CHECKBOX_CONTROL_VALUE_ACCESSOR],
+    encapsulation: ViewEncapsulation.None,
+})
+export class CheckBoxComponent implements ControlValueAccessor, OnInit {
+    @Input() label: string | TemplateRef<HTMLElement>;
+    @Input() _id?: string;
```

Comment:
I think we had run though an issue in the past where using `id` caused issues since it renders on the custom html element like this `<app-checkbox id="myId"></app-checkbox>`. Querying the id will return two elements with the same `id`. That's why we decided to use `_id` because that will be considered a custom property and will be passed to the element.

Reply:
There is way around this, but i will have to pass id as an expression.
```HTML
<app-checkbox [id]="'myId'"></app-checkbox>
```
would result in
```HTML
<app-checkbox ng-reflect-id="myId" >
    <div id="myId"></div>
</app-checkbox>
```
This would solve the extra id property created but in the other hand will force the user to write that expression syntax